client/state/plugins/premium/reducer.js

```diff
@@ -0,0 +1,167 @@
+/**
+* External dependencies
+*/
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PLUGIN_SETUP_FETCH_INSTRUCTIONS,
+	PLUGIN_SETUP_RECEIVE_INSTRUCTIONS,
+	PLUGIN_SETUP_START,
+	PLUGIN_SETUP_INSTALL,
+	PLUGIN_SETUP_ACTIVATE,
+	PLUGIN_SETUP_CONFIGURE,
+	PLUGIN_SETUP_FINISH,
+	PLUGIN_SETUP_ALL_FINISH,
+	PLUGIN_SETUP_ERROR
+} from 'state/action-types';
+
+/*
+ * Tracks the requesting state for premium plugin "instructions" (the list
+ * of plugins and API keys) on a per-site index.
+ */
+export function isRequesting( state = false, action ) {
+	switch ( action.type ) {
+		case PLUGIN_SETUP_FETCH_INSTRUCTIONS:
+			return Object.assign( {}, state, { [ action.siteId ]: true } );
+		case PLUGIN_SETUP_RECEIVE_INSTRUCTIONS:
+			return Object.assign( {}, state, { [ action.siteId ]: false } );
+		default:
```

Comment:
I should add a warning for this, but each reducer should also implement `SERIALIZE` and `DESERIALIZE` (see [data persistence](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#data-persistence--2754-) ) Without these cases, this subtree will be persisted by default. If you're not ready to persist state here, return the default state, which in this case is `false`.